src/sentry/tasks/summaries/weekly_reports.py

```diff
@@ -198,17 +136,59 @@ def prepare_organization_report(
     with sentry_sdk.start_span(op="weekly_reports.user_project_ownership"):
         user_project_ownership(ctx)
     with sentry_sdk.start_span(op="weekly_reports.project_event_counts_for_organization"):
-        project_event_counts_for_organization(ctx)
+        event_counts = project_event_counts_for_organization(ctx)
+        for data in event_counts:
+            project_id = data["project_id"]
+            # Project no longer in organization, but events still exist
+            if project_id not in ctx.projects:
+                continue
+            project_ctx = ctx.projects[project_id]
+            total = data["total"]
+            timestamp = int(to_timestamp(parse_snuba_datetime(data["time"])))
+            if data["category"] == DataCategory.TRANSACTION:
+                # Transaction outcome
+                if data["outcome"] == Outcome.RATE_LIMITED or data["outcome"] == Outcome.FILTERED:
+                    project_ctx.dropped_transaction_count += total
+                else:
+                    project_ctx.accepted_transaction_count += total
+                    project_ctx.transaction_count_by_day[timestamp] = total
+            elif data["category"] == DataCategory.REPLAY:
+                # Replay outcome
+                if data["outcome"] == Outcome.RATE_LIMITED or data["outcome"] == Outcome.FILTERED:
+                    project_ctx.dropped_replay_count += total
+                else:
+                    project_ctx.accepted_replay_count += total
+                    project_ctx.replay_count_by_day[timestamp] = total
+            else:
+                # Error outcome
+                if data["outcome"] == Outcome.RATE_LIMITED or data["outcome"] == Outcome.FILTERED:
+                    project_ctx.dropped_error_count += total
+                else:
+                    project_ctx.accepted_error_count += total
+                    project_ctx.error_count_by_day[timestamp] = (
+                        project_ctx.error_count_by_day.get(timestamp, 0) + total
+                    )
 
     with sentry_sdk.start_span(op="weekly_reports.organization_project_issue_substatus_summaries"):
         organization_project_issue_substatus_summaries(ctx)
 
     with sentry_sdk.start_span(op="weekly_reports.project_passes"):
         # Run project passes
         for project in organization.project_set.all():
-            project_key_errors(ctx, project)
+            key_errors = project_key_errors(ctx, project)
+            if key_errors:
+                ctx.projects[project.id].key_errors = [
+                    (e["group_id"], e["count()"]) for e in key_errors
+                ]
+                if ctx.organization.slug == "sentry":
+                    logger.info(
+                        "project_key_errors.results",
+                        extra={"project_id": project.id, "num_key_errors": len(key_errors)},
+                    )
             project_key_transactions(ctx, project)
```

Comment:
Is this one still modifying in place?
can we update to return also?

Reply:
It is - I didn't update it because I don't have plans to use this function again, but I can for consistency.

Reply:
Updated - I had to break it out into 2 functions FYI